### DISCUSSION TITLE
Help：How can LLM output be adapted to JSON format?

### DISCUSSION THREAD

**Discussion by bakva9** [langgenius/dify#10678] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

When I try to send LLM output variables as JSON format from HTTP request to GAS (GoogleAppScript), I get an error that LLM output is not in JSON format, the JSON format I write in the HTTP request node is correct (I checked many times). I think the problem is in the LLM output itself, but how can I make it conform to the JSON format?

**Comment by akira0912** [comment]
Hi, did you find a solution?

**Comment by leeey111** [comment]
```json
    {
      "role": "system",
      "content": "****your prompts. Return a JSON response only. IMPORTANT: Format the output as a JSON. Only return a JSON response with no other comment or text. If you return any other text than JSON, you will have failed.",
    }
```
i use this prompt, and try it on deepseek and qwen, it works well.

**Reply by akira0912** [reply]
thank you.I also solved my problem.

**Comment by breakstring** [comment]
For GPT-4o, it supports constraining output to a JSON structure using a json_schema. However, it's unclear if this is supported in Dify, as it requires beforehand constructing the schema or defining it through Pydantic's BaseModel. If not supported, especially in UI-based workflows like Dify that encapsulate many details, I suggest:

1. Clearly provide the structure and output examples in the prompt.
2. Validate the result afterward and extract the needed data.
